workers/unity/Assets/Playground/Scripts/Cubes/ScoreCollision.cs

```diff
@@ -0,0 +1,47 @@
+using Generated.Playground;
+using Improbable.Gdk.Core;
+using Unity.Entities;
+using UnityEngine;
+
+public class ScoreCollision : MonoBehaviour
+{
+    private MutableView view;
+    private EntityManager entityManager;
+
+    public void Start()
+    {
+        var component = GetComponent<SpatialOSComponent>();
+        if (component)
+        {
+            view = WorkerRegistry.GetWorkerForWorld(component.World).View;
+            entityManager = component.World.GetExistingManager<EntityManager>();
+        }
+        else
+        {
+            Debug.LogError("Could not retrieve SpatialOSComponent from game object.");
+        }
+    }
+
+    public void OnCollisionEnter(Collision col)
+    {
+        if (entityManager == null)
+        {
+            return;
+        }
+
+        if (col.gameObject && col.gameObject.tag == "Cube")
+        {
+            var component = GetComponent<SpatialOSComponent>();
+            if (component != null && view.HasComponent(component.Entity, typeof(SpatialOSLaunchable))
+                && !view.HasComponent(component.Entity, typeof(Playground.CollisionComponent)))
```

Comment:
you only use the `MutableView` for `HasComponent`. Remove the `MutableView` completely from this script and use the `entityManager` directly. If you look at the code, we only use it as a wrapper and is really not necessary.

Reply:
Good point! Doing so right now.